### ISSUE TITLE
Restart on crash - or add some redundancy.

### ISSUE THREAD

**Issue by alexmorley** [command-line-bootcamp/command_line_bootcamp#33]
to prevent e.g. #32

**Comment by blahah** [comment]
Yeah, I'm not sure why this is happening but I think it's to do with docker container resource management. There is probably a systematic solution out there somewhere but I haven't discovered it yet. I might ping some experts on twitter this weekend.

**Comment by alexmorley** [comment]
Hmm slightly different problem this time, got a bug report on twitter. But same solution might apply. For now just restarting the VM works OK.

> Hi, I'm a CS professor trying to use http://rik.smith-unna.com/command_line_bootcamp/ … for a lab tomorrow, but I can't type in the shell area and don't get a prompt (just a flashing cursor). I've tried this on multiple machines and in multiple browsers on Windows (FF, Chrome, IE).

**Comment by alexmorley** [comment]
I have added an hourly script to clean up empty containers and do some logging to make it easier to work out what's going on. I suspect it's RAM. In which case I will try add some more to the VM ...

**Comment by blahah** [comment]
nice @alexmorley - can you add the script to the repo?

**Comment by alexmorley** [comment]
@blahah Sorry missed this. yes, but I also have a better solution going forward.

**Comment by alexmorley** [comment]
Proposal: Add a (very) lightweight orchestrating server between the front-end and docker-browser-server. It only needs a single API entrypoint.

### Orchestrator
`GET /url?container_id`
```json
{
  "createdAt": "2011-09-20T01:02:17.931Z",
  "ip_addr": "8.8.8.8"
}
```

### Docker-Browser-Server
Ideally we would also add the another entry-point to the docker-browser-server API that returns some memory/cpu availability statistics (either overall or per container).
`GET /stats?container_id`
and
`GET /stats`
```json
{
  "createdAt": "2011-08-20T02:06:57.931Z",
  "cpu_percentage": "88",
  "mem_percentage": "88",
  "mem_free": "1028"
}
```

**Comment by alexmorley** [comment]
working on branch [orchestra](https://github.com/blahah/command_line_bootcamp/tree/orchestra)
